### PR TITLE
Move smart contract test files to base

### PR DESCRIPTION
**Pull request by Bargsteen**
## Purpose

Move the smart contract test files to concordium base since they are needed in the smart contract integration testing library.

Depends on:
- https://github.com/Concordium/concordium-base/pull/332

## Changes

- Move the files.
- Update references to the files.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-node/Cargo.lock [hunk 1]
```diff
@@ -1179,7 +1179,6 @@ dependencies = [
  "rand 0.7.3",
  "random_oracle",
  "serde 1.0.145",
- "wasm-bindgen",
```
Comment (Bargsteen) [inline concordium-node/Cargo.lock:1182]: I didn't make any changes in `Cargo.toml`, so I assume we just forgot to push the lock file recently.

Comment (limemloh) [reply concordium-node/Cargo.lock:1182]: It might be connected to the conflicts in base.
Try updating it and update the lock file

Comment (abizjak) [reply concordium-node/Cargo.lock:1182]: I think the lock files are up-to-date as long as your base is. It looks like maybe your base is not up-to-date?

This was changed recently (like yesterday).

**Review by abizjak — COMMENTED** [review]
Conflicts need to be resolved. I think first in base.

**Review by abizjak — APPROVED** [review]
Looks good.

**Pull request merged**